File: sort/bogo_sort.py
import random
from typing import List

def in_order(numbers: List[int]) -> bool:
  for i in range(len(numbers)-1):
    if numbers[i] > numbers[i+1]:
      return False
    
  return True

  # all(numbers[i] <= numbers[i+1] for i in range(len(numbers)-1))

def bogo_sort(numbers: List[int]) -> List[int]:
  # print(numbers)
  while not in_order(numbers):
    random.shuffle(numbers)
  return numbers

if __name__ == '__main__':
  li = [1,3,4,2,5]
  print(li)
  print(bogo_sort(li))

File: golang/sort/bogo_sort.go
package main

import (
	"fmt"
	"math/rand"
)

func BogoSort(numbers []int) []int {
	for !inOrder(numbers) {
		shuffle(numbers)
	}
	return numbers
}


func inOrder(numbers []int) bool {
	for i := 0; i < len(numbers) - 1; i++ {
		if numbers[i] > numbers[i+1] {
			return false
		}
	}
	return true
}

func shuffle(a []int) []int {
	for i := range a {
		j := rand.Intn(i + 1)
		a[i], a[j] = a[j], a[i]
	}
	return a
}

func main() {
	li := []int{1,3,2,6,5}
	fmt.Println(li)
	fmt.Println(BogoSort(li))
}